lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -2135,6 +2184,15 @@ instance (forall era. IsCardanoEra era => Eq (thing era)) =>
         Just Refl -> a == b
         Nothing -> False
 
+parseTxHash :: TxCBOR -> Either DeserialiseFailure (W.Hash "Tx")
+parseTxHash cbor = parseCBOR cbor <&> \case
+    EraTx ByronEra x -> byronTxHash x
+    EraTx ShelleyEra x -> shelleyTxHash x
+    EraTx MaryEra x -> shelleyTxHash x
+    EraTx AllegraEra x -> shelleyTxHash x
+    EraTx AlonzoEra x -> shelleyTxHash x
+    EraTx BabbageEra x -> shelleyTxHash x
+
```

Comment:
```suggestion
```

Ah! I would put this function next to the definition of the `TxCBOR` type, as its part of its interface / abstraction boundary. 🤓 

If the import spaghetti between `core` and `shelley` becomes too much, I recommend putting `TxCBOR` into a separate module `Cardano.Wallet.Primitive.Types.TxCBOR` and define `byronTxHash` etc. in that module.